### PR TITLE
Internal: Infra for the ability to define page transitions [ED-3849]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: includes/base/controls-stack.php [hunk 1]
```diff
@@ -137,6 +137,18 @@ abstract class Controls_Stack extends Base_Object {
...
+	 *
+	 * Holds all the render attributes of the element. Used to store data like
+	 * the HTML class name and the class value, or HTML element ID name and value.
+	 *
+	 * @access private
+	 *
+	 * @var array
+	 */
+	private $render_attributes = [];
+
```
Comment (StyleShit) [inline includes/base/controls-stack.php:151]: All those "new" functions here have just been moved up from `element-base.php`, don't bother reading them.

File: includes/managers/controls.php [hunk 2]
```diff
@@ -915,6 +915,51 @@ public function add_custom_css_controls( Controls_Stack $controls_stack, $tab =
 		$controls_stack->end_controls_section();
 	}
 
+	/**
+	 * Add Page Transitions controls.
+	 *
+	 * This method adds a new control for the "Page Transitions" feature. The free
```
Comment (matipojo) [inline includes/managers/controls.php:921]:
```suggestion
	 * This method adds a new control for the "Page Transitions" feature. The core
```

File: core/kits/documents/tabs/settings-page-transitions.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	public function get_icon() {
+		return 'eicon-layout-settings';
+	}
+
+	public function get_help_url() {
+		return 'https://go.elementor.com/page-transitions';
+	}
+
+	protected function register_tab_controls() {
+		Plugin::$instance->controls_manager->add_page_transitions_controls( $this->parent, $this->get_id() );
```
Comment (matipojo) [inline core/kits/documents/tabs/settings-page-transitions.php:35]: If we don't plan to add the ability to override the settings per page - the controls should be here

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/managers/controls.php (hunk 2 above)
Comment (StyleShit) [reply includes/managers/controls.php:921]: Done.

File: core/kits/documents/tabs/settings-page-transitions.php (hunk 3 above)
Comment (StyleShit) [reply core/kits/documents/tabs/settings-page-transitions.php:35]: There are plans to do that in the future.

File: core/kits/documents/kit.php [hunk 4]
```diff
@@ -51,6 +51,17 @@ public function get_tabs() {
...
 
+	/**
+	 * Retrieve a tab by ID.
+	 *
+	 * @return Tabs\Tab_Base
+	 */
+	public function get_tab( $id ) {
+		$tabs = $this->get_tabs();
+
+		return isset( $tabs[ $id ] ) ? $tabs[ $id ] : null;
```
Comment (kobizz) [inline core/kits/documents/kit.php:62]: You can simply do `return self::get_items( $this->get_tabs(), $id );`

**Review by matipojo — APPROVED** [review]

**Review by kobizz — CHANGES_REQUESTED** [review]

File: core/kits/documents/kit.php (hunk 4 above)
Comment (StyleShit) [reply core/kits/documents/kit.php:62]: Done.

**Review by matipojo — APPROVED** [review]

**Pull request merged**